src/fetch_request.js

```diff
@@ -93,6 +93,24 @@ export class FetchRequest {
     return this.options.body
   }
 
+  get query () {
+    const originalQuery = (this.originalUrl.split('?')[1] || '').split('#')[0]
+    const params = new URLSearchParams(originalQuery)
+
+    if (this.options.query) {
+      for (const [key, value] of Object.entries(this.options.query)) {
+        params.append(key, value)
+      }
+    }
+
+    const query = params.toString()
+    return (query.length > 0 ? `?${query}` : '')
+  }
+
+  get url () {
+    return this.originalUrl.split('?')[0] + this.query
```

Comment:
This probably wouldn't matter for ajax calls, but a side effect of this is if the url has an anchor in it (for whatever reason), it would be removed.

Reply:
Yeah, I didn't see any reason to keep the anchor for AJAX requests. I don't know of any use case for it server-side actually.